File: backend/app/schemas/asignaciones.py
"""Pydantic schemas for Asignacion entity."""

import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class AsignacionCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")
    usuario_id: uuid.UUID
    rol: str
    materia_id: uuid.UUID | None = None
    carrera_id: uuid.UUID | None = None
    cohorte_id: uuid.UUID | None = None
    comisiones: list[str] | None = None
    responsable_id: uuid.UUID | None = None
    desde: date
    hasta: date | None = None


class AsignacionUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")
    rol: str | None = None
    materia_id: uuid.UUID | None = None
    carrera_id: uuid.UUID | None = None
    cohorte_id: uuid.UUID | None = None
    comisiones: list[str] | None = None
    responsable_id: uuid.UUID | None = None
    desde: date | None = None
    hasta: date | None = None


class AsignacionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True, extra="forbid")
    id: uuid.UUID
    tenant_id: uuid.UUID
    usuario_id: uuid.UUID
    rol: str
    materia_id: uuid.UUID | None
    carrera_id: uuid.UUID | None
    cohorte_id: uuid.UUID | None
    comisiones: list[str]
    responsable_id: uuid.UUID | None
    desde: date
    hasta: date | None
    estado_vigencia: str
    created_at: datetime
    updated_at: datetime


__all__ = [
    "AsignacionCreate",
    "AsignacionUpdate",
    "AsignacionResponse",
]

File: frontend/src/features/gestion-comision/types/asignaciones.ts
/**
 * types/asignaciones.ts — TypeScript types for asignacion selector.
 *
 * Source of truth: backend/app/schemas/asignacion.py
 * No `any` allowed.
 */

export type EstadoVigencia = 'Vigente' | 'Vencida'

export interface AsignacionResponse {
  id: string
  tenant_id: string
  usuario_id: string
  rol: string
  materia_id: string | null
  carrera_id: string | null
  cohorte_id: string | null
  comisiones: string[]
  responsable_id: string | null
  desde: string
  hasta: string | null
  estado_vigencia: EstadoVigencia
  created_at: string
  updated_at: string
}
